### PR TITLE
Implement class ComputerCourseState and subclasses

### PR DESCRIPTION
**Pull request by yevGen-S**
**Add abstract class ComputerCourseState**

-  add subclass NullComputerCourseState
-  add subclass OnlineWorkCourseState
-  add subclass WebDesignCourseState
-  add subclass VideoEditingCourseState
-  add subclass GameDevelopmentCourseState
   Closes: #3

### CODE REVIEW

**Comment by yevGen-S** [comment]
Please add prices for courses

**Comment by yevGen-S** [comment]
And sorry for again pulling request with Kirill's branch, I'll fix it.

**Review by Bazalii — CHANGES_REQUESTED** [review]
You mustn't use your previous branch with all the files here. Only ComputerCourseState files should be in this request because all the other files will be viewed in the other PR and it is hard to track duplicates. I think, that you didn't checkout to master to make a new branch and just made it, staying on your previous feature branch. Is is bad because all the history from your previous branch is here. There are two ways in this situation. First - you can rewrite history here to leave commits that are connected with this PR. The other is to delete this branch and make PR closed after that make a new branch correctly and new PR. I recommend to use the first one because the second one isn't clean, this PR will remain in closed section but it isn't useful at all, so try to rewrite the history!

**Comment by yevGen-S** [comment]
fix: rewrite history of commits

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package android.clicker.school_live_simulator
+
+abstract class ComputerCourseState: StudyCourseState {
+    protected abstract val price: Int
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:4]: Please add commentary about price

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * This is a list of available web tasks
+     * These types are described in enumeration class WebTask
+     */
+
+    protected val available_web_task: List<WebTask> = listOf<WebTask>()
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:11]:
```suggestion
    protected val available_web_tasks: List<WebTask> = listOf<WebTask>()
```

Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:11]: @Ferrociraptus, what should we use - list or arraylist?

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * This is a function for buying courses
+     * @param student Player who buys course
+     */
+
+    abstract fun buyNextCourse(student: Player)
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:18]: Same thing for every class
```suggestion
    abstract fun buyNextCourse(courses: Player.Courses)
```

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+open class NullComputerCourseState: ComputerCourseState {
+    override price: Int = 0
+
+    init {
+        available_web_task.add(WebTask.ACCOUNT_BOOST)
+    }
+
+    override fun buyNexCourse(student: Player) {
+        Player.courses = OnlineWorkCourseState()
+    }
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt:12]: Same thing for every class
```suggestion
    override fun buyNexCourse(courses: Player.Courses) {
        courses.computer_course = OnlineWorkCourseState()
    }
```

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package android.clicker.school_live_simulator
+
+open class NullComputerCourseState: ComputerCourseState {
+    override price: Int = 0
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt:4]: @Ferrociraptus, There is no information in ClickerUPD about prices for courses...

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    /**
+     * This is a function to ckeck if course is available to buy
+     * @param web_task Element of enum WebTask
+     * @return True if course is available to buy and False if not
+     */
+
+    fun isAvailable(web_task: WebTask): Boolean {
+        return web_task in available_web_task
+    }
+
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:29]: Where are timerTickHandler() and timerEndHandler()?

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt (hunk 5 above)
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt:4]: Same thing in every class
```suggestion
    override val price: Int = 0
```

**Review by Bazalii — CHANGES_REQUESTED** [review]
Just check all my comments, don't forget to correct mistakes in every class...

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt (hunk 1 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:9]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt (hunk 2 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt (hunk 3 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt (hunk 6 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:34]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt (hunk 5 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt]: Fixed

Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt]: What to do with prices of courses? Inform me please when they will be added in ClickerUPD and I'll add them.

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt (hunk 4 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt (hunk 5 above)
Comment (Bazalii) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt]: > What to do with prices of courses? Inform me please when they will be added in ClickerUPD and I'll add them.

You can now give them zero values and just add new issue like Andrey did to add them in the future

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt (hunk 2 above)
Comment (Bazalii) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt]: @yevGen-S Use here arrayList instead of List pls

Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt (hunk 5 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt]: Fixed

**Review by Bazalii — CHANGES_REQUESTED** [review]
Several things should be corrected, also add one more child class FriendsCourse

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt [hunk 7]
```diff
@@ -0,0 +1,35 @@
+package android.clicker.school_live_simulator
+
+abstract class ComputerCourseState: StudyCourseState {
```
[the remark this answers is not in the document]
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package android.clicker.school_live_simulator
+
+open class NullComputerCourseState: ComputerCourseState {
+    override val price: Int = 0
+
+    init {
+        available_web_task.add(WebTask.ACCOUNT_BOOST)
+    }
```
[the remark this answers is not in the document]
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/NullComputerCourseState.kt]: Fixed

**Review by Bazalii — CHANGES_REQUESTED** [review]
Everything is nice but timerTickHandler() and timerEndHandler() should be realised. I or @Ferrociraptus will write about these two methods

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt [hunk 9]
```diff
@@ -0,0 +1,35 @@
+package android.clicker.school_live_simulator
+
+abstract class ComputerCourseState: StudyCourseState() {
+
+    /**
+     * This is a price of computer course
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt:6]: I think it is better to remove this is in every commentary. Just _price of computer_ , _list of smth_

**Review by Bazalii — CHANGES_REQUESTED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt [hunk 10]
```diff
@@ -0,0 +1,16 @@
+package android.clicker.school_live_simulator
+
+open class OnlineWorkCourseState : NullComputerCourseState() {
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt:3]: parent whould be FriendsCourseState

**Review by Bazalii — CHANGES_REQUESTED** [review]
There is only one thing left

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt (hunk 10 above)
Comment (Bazalii) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt]: Fixed for faster review

**Review by Bazalii — APPROVED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt (hunk 10 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt:3]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt (hunk 9 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/ComputerCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt (hunk 10 above)
Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/OnlineWorkCourseState.kt]: Fixed

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/FriendsCourseState.kt [hunk 11]
```diff
@@ -0,0 +1,16 @@
+package android.clicker.school_live_simulator
+
+open class FriendsCourseState : NullComputerCourseState() {
+    override val price: Int = 0
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/FriendsCourseState.kt:4]: @yevGen-S, please, talk with @Bazalii and set base price value in fields with same names.
It should be any values, but would be better if prices values will be organized in arithmetic or geometric progression.

**Review by Ferrociraptus — CHANGES_REQUESTED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/FriendsCourseState.kt (hunk 11 above)
Comment (Bazalii) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/FriendsCourseState.kt:4]: Okay, lets add values from friend's course and let them be: 100, 300, 900, 2700, 8100

Comment (yevGen-S) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/FriendsCourseState.kt:4]: Fixed

**Review by Bazalii — APPROVED** [review]

**Review by Ferrociraptus — APPROVED** [review]

**Pull request merged**